test/orm/test_utils.py

```diff
@@ -1162,3 +1168,70 @@ def test_with_poly_use_mapper(self):
         # polymorphic AliasedClass with the "use_mapper_path" flag -
         # the AliasedClass acts just like the base mapper
         eq_(p1.path, (emapper, emapper.attrs.machines))
+
+
+class PolymorphicWithSelectableAndFlatTest(fixtures.DeclarativeMappedTest):
```

Comment:
so this is good, but what I want to do here is make this test use a lot fewer APIs.   I guess the test here is making use of the full test case for #4212, but for testing the actual thing here we can do something much more minimal, and it can go in test/orm/inheritance/test_with_poly.py and make use of the existing fixtures:

```

diff --git a/test/orm/inheritance/test_with_poly.py b/test/orm/inheritance/test_with_poly.py
index 3ff6c191c..fc8cea0e8 100644
--- a/test/orm/inheritance/test_with_poly.py
+++ b/test/orm/inheritance/test_with_poly.py
@@ -1,4 +1,5 @@
 from sqlalchemy import and_
+from sqlalchemy import exc
 from sqlalchemy import or_
 from sqlalchemy import testing
 from sqlalchemy.orm import create_session
@@ -16,6 +17,19 @@ from ._poly_fixtures import Manager
 from ._poly_fixtures import Person
 
 
+class WithPolymorphicAPITest(_Polymorphic, _PolymorphicFixtureBase):
+    def test_no_use_flat_and_aliased(self):
+        sess = create_session()
+
+        subq = sess.query(Person).subquery()
+
+        testing.assert_raises_message(
+            exc.ArgumentError,
+            "Argument cannot be selectable and flat",
+            with_polymorphic, Person, [Engineer],
+            selectable=subq, flat=True
+        )
+
 class _WithPolymorphicBase(_PolymorphicFixtureBase):
     def test_join_base_to_sub(self):
         sess = create_session()

```

Reply:
noted, much appreciated!